Sources/Bot/Services/MergeService.swift

```diff
@@ -356,11 +359,42 @@ extension MergeService {
         })
     }
 
-    fileprivate static func whenStarting(initialPullRequests: [PullRequest], scheduler: DateScheduler) -> Feedback<State, Event> {
+    fileprivate static func whenStarting(
+        initialPullRequests: [PullRequest],
+        gitHubAPI: GitHubAPIProtocol,
+        botUser: GitHubUser?,
+        scheduler: DateScheduler
+    ) -> Feedback<State, Event> {
+        typealias DatedPullRequest = (pr: PullRequest, date: Date)
+
+        func isBotMergeComment(_ comment: IssueComment) -> Bool {
+            return (botUser.map({ comment.user.id == $0.id }) ?? true) // If botUser nil, default to true
+                && comment.body.hasPrefix(MergeService.acceptedCommentIntro)
+        }
+
+        func datedPullRequest(for pullRequest: PullRequest) -> SignalProducer<DatedPullRequest, Never> {
+            return gitHubAPI.fetchIssueComments(in: pullRequest)
+                .flatMapError { _ in .value([]) }
+                .map { comments in
+                    let lastBotComment = comments
+                        .filter(isBotMergeComment)
+                        .max { $0.creationDate < $1.creationDate }
+                    return (pullRequest, lastBotComment?.creationDate ?? .distantFuture)
```

Comment:
Would this work?

```suggestion
                        .filter(isBotMergeComment)
                        .map(\.creationDate)
                        .min()
                    return (pullRequest, lastBotComment ?? .distantFuture)
```

Reply:
The goal is to find the max (most recent) date. I _think_ `max(<)` is semantically "order by `<` aka from smallest to biggest, then take the max value" aka "get the biggest date value", while `min(date)` would give us the smallest (aka oldest) date…

Makes me doubt though, I'll double-check the expected meaning of the parameter to `max`

Reply:
Updated code to use `.map` (couldn't use KeyPath since bot is not yet on latest Swift version) + `.max()`

Also updated the test to add a trap with a bot comment in Jan in the later PR before the later comment in March

See 8b6a72d20ab8f5cb6cd6d6bf9300bb14fdb9e681